R/backend-teradata.R

```diff
@@ -109,8 +109,11 @@ sql_translation.Teradata <- function(con) {
                         digits <- vctrs::vec_cast(digits, integer())
                         sql_expr(CAST(!!x %as% DECIMAL(12L, !!digits)))
                       },
-      as.double     = sql_cast("NUMERIC"),
-      as.character  = sql_cast("VARCHAR(MAX)"),
+      as.double     = sql_cast("FLOAT"),
+      as.character  = function(x, nchar = 255L) {
+        nchar <- vctrs::vec_cast(nchar, integer())
```

Comment:
This doesn't feel quite right because I think you want `nchar` to be a scalar? So `check_number_whole()` would be more appropriate here (but I don't know if we have that in dbplyr yet). @simonpcouch should be able to help (otherwise this PR looks good to me at least within the limitation of my SQL knowledge 😄)